### ISSUE TITLE
Save odometer value so that it wont be lost after shutoff by battery switch

### ISSUE THREAD

**Issue by jnkr1** [OpenSource-EBike-firmware/Color_LCD#66]
The odometer currently gets saved when shutting down the 850c by long-press. However when cutting the power off by battery switch, the odometer doesnt get saved. I propose to save the odometer value (and other future similarly persistent values) using a 10 (or some other arbitrary amount) second interval.

Use case described here: https://www.endless-sphere.com/forums/viewtopic.php?f=30&t=93818&start=4450#p1525665

**Comment by Gabij** [comment]
I was wondering the same.
According to the soc datasheet, the flash memory is rated for 20k writes.
But I'm not sure how this is counted. See 
https://devzone.nordicsemi.com/f/nordic-q-a/15798/flash-endurance-on-nrf52

Saving every 10s is not realistic.
Some ideas:
Every miles/km
Every time the bike stop
Detect the power loss throught the Vbatt sensor and save odometer value before the shutdown. Need to be tested if there is enought time without hardware modifications

**Comment by jnkr1** [comment]
Yes there's a limitation. This might be a feasible way to do it:

is the bike stopped, yes --> is the saved odo different from the actual odo (difference by 0.1 mile or more), yes --> save values

**Comment by jnkr1** [comment]
Bump - please address this issue

**Comment by livello** [comment]
20,000 stops? And still you have a chance to loose your settings. 
I think the way out is to pair sw102 with smartphone. So android app would store all necessary information.
